contracts/Commitment/Commitment.sol

```diff
@@ -210,22 +262,57 @@ contract Commitment is
         payable
         withTimedTransitions()
         onlyStates(State.Whitelist, State.Public)
+
+        // AUDIT[CHF-55] Accepting agreement confusion.
+        //   This will automatically put the msg.sender in the list of addresses
+        //   accepting the agreement. Shouldn't the order of actions be
+        //   different? Accepting the agreement being a precondition of commit()
+        //   action?
         acceptAgreement(msg.sender)
     {
         // Take with EtherToken allowance (if any)
         uint256 commitedWei = ETHER_TOKEN.allowance(msg.sender, this);
+
+        // AUDIT[CHF-54] Unnecessary call to ETHER_TOKEN.transferFrom().
+        //   When commitedWei is 0, calling ETHER_TOKEN.transferFrom() will
```

Comment:
There is minimum ticket size which is 1 ETH now, so in the case you describe contract will revert. Surely, gas will be used but that's going to be user's fault. Blockchain will not be polluted with unnecessary log entries.

It may be good idea to move minimum ticket checks before transfers and deposits are made: move it from `commitToken` and duplicat in `commit` and `commitEuro`. That would make contract clear.